app/components/forms/wizard/sessions-speakers-step.js

```diff
@@ -175,19 +175,35 @@ export default Component.extend(EventWizardMixin, FormMixin, {
   },
 
   actions: {
-    addItem(type, position) {
+    addItem(type) {
       switch (type) {
         case 'sessionType':
           this.data.sessionTypes.addObject(this.store.createRecord('session-type'));
           break;
         case 'track':
           this.data.tracks.addObject(this.store.createRecord('track'));
           break;
-        case 'microlocation':
-          this.data.microlocations.addObject(this.store.createRecord('microlocation', { position }));
-          break;
       }
     },
+
+    addRoom(index) {
+      this.microlocations.forEach(room => {
+        const pos = room.get('position');
+        pos > index && room.set('position', pos + 1);
+      });
```

Comment:
Why is there a need of adding this complex logic?

Reply:
As you can see from interacting process, if you add a new room, it will be added after the room which is clicked to insert new room.
Above code perform like this process.
To illustrate above process, suppose there are 4 rooms and their positions are 1, 2, 3, 4.
And then if click plus(+) button on the 2nd room, inserted new room's position is 3 and previous 3th and 4th rooms should be 4th and 5th.
That is all of my reason.